src/io/flutter/jxbrowser/JxBrowserManager.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2020 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.jxbrowser;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.progress.impl.BackgroundableProcessIndicator;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Pair;
+import com.intellij.util.containers.ContainerUtil;
+import com.intellij.util.download.DownloadableFileDescription;
+import com.intellij.util.download.DownloadableFileService;
+import com.intellij.util.download.FileDownloader;
+import com.intellij.util.lang.UrlClassLoader;
+import io.flutter.utils.FileUtils;
+import io.flutter.utils.JxBrowserUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+enum JxBrowserStatus {
+  NOT_INSTALLED,
+  INSTALLATION_IN_PROGRESS,
+  INSTALLED,
+  INSTALLATION_FAILED,
+}
+
+// JxBrowser provides Chromium to display web pages within IntelliJ. This class manages downloading the required files and adding them to
+// the class path.
+public class JxBrowserManager {
+  private static JxBrowserManager manager;
+  private static final String DOWNLOAD_PATH = FileUtils.platformPath();
+  private static final AtomicReference<JxBrowserStatus> status = new AtomicReference<>(JxBrowserStatus.NOT_INSTALLED);
+  private static final Logger LOG = Logger.getInstance(JxBrowserManager.class);
+  // We will be gating JxBrowser features until all of the features are landed.
+  private static final boolean ENABLE_JXBROWSER = false;
+
+  private JxBrowserManager() {}
+
+  public static JxBrowserManager get() {
+    if (manager == null) {
+      return new JxBrowserManager();
+    }
+    return manager;
+  }
+
+  public boolean isInstalled() {
+    return status.get().equals(JxBrowserStatus.INSTALLED);
+  }
+
+  public void setUp(Project project) {
+    if (!ENABLE_JXBROWSER) {
```

Comment:
Should be ENABLE_JX_BROWSER not ENABLE_JXBROWSER. Also move this check to outside this class to where setup is called. That way it is obvious by looking at that code that we aren't doing anything unless the flag is set.